### PR TITLE
[doc] Rounding range query rules

### PR DESCRIPTION
**Pull request by pgomulka**
I added this in range query seciton, but maybe it should be in date math section?
https://elasticsearch_63109.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/common-options.html#date-math
relates https://github.com/elastic/elasticsearch/issues/62268

### CODE REVIEW

**Review by jrodewig — APPROVED** [review]
LGTM. I relocated your section and added some xrefs. I also removed an outdated warning as we discussed.

Thanks for taking care of this.

File: docs/reference/mapping/types/date.asciidoc [hunk 1]
```diff
@@ -91,6 +91,30 @@ PUT my-index-000001
...
+
+[source,text]
+----
+MONTH_OF_YEAR:    01
+DAY_OF_MONTH:     01
+HOUR_OF_DAY:      23
+MINUTE_OF_HOUR:   59
+SECOND_OF_MINUTE: 59
+NANO_OF_SECOND:   999_999_999
+----
```
Comment (jrodewig) [inline docs/reference/mapping/types/date.asciidoc:111]: Are these values only applied to `gt` or `gte` values or do they apply to `lt` and `lte` as well?

**Comment by jrodewig** [comment]
On second thought, I decided to leave this on the range query page as its really dealing with date values provided in a query or aggregation rather than `date` fields.

**Comment by pgomulka** [comment]
wow! that really looks good now. Thanks @jrodewig for this.

**Pull request merged**